### ISSUE TITLE
Can new repository creating be automated?

### ISSUE THREAD

**Issue by berndpfrommer** [ros2-gbp/ros2-gbp-github-org#600]
Creating a new release repository sometimes happens within a few days, in other cases can take more than a week (my most recent issue has been sitting only for 5 days, so it's still an infant at this point).
I understand there is some manual process involved and the maintainers are busy, but wouldn't it make sense to automate this process? It does kinda disrupt my development flow when I have a step that can take from a few hours to a few weeks, and you don't know when it'll happen. Plus, in the long run you will recoup the development time by not having to manually maintain a repo anymore.

**Comment by nuclearsandwich** [comment]
> I understand there is some manual process involved and the maintainers are busy, but wouldn't it make sense to automate this process?

The short answer is yes. It does make sense. We've talked about wiring this up with ros/rosdistro so that getting included in distribution.yaml automatically creates the release team and repositories.
This isn't a slam dunk because there's missing data in the upstream sources: specifically there's no association between a GitHub username and a maintainer email address in package.xml.

Figuring out how and where best to add this information to the distribution.yaml specification is the main sticking point that prevents this from being a priority for the infrastructure project.

**Comment by berndpfrommer** [comment]
OK, I'll leave this issue open then as a place holder. No offense if you close it with "Won't Fix" :)

**Comment by Timple** [comment]
Well, that's a far fetched goal.

But perhaps easier would be an automation script. 

Create a text file in this repository where people can add their desired repository.
If a new entry is added an action can create the repository based on the template and the given name.

Now your work went from manual repository creation to clicking the merge button. And that's something you can also easily trust other with when you're on holiday as there is no learning involved.

**Comment by nuclearsandwich** [comment]
> But perhaps easier would be an automation script.

In ROS PMC meetings we've discussed moving repository maintainer information into ros/rosdistro and dramatically simplifying this project into a backend implementation detail derived from information stored in the rosdistro repo. The most painful part of this process is the bouncing back and forth between ros2-gbp and rosdistro. If everything is in rosdistro the review (and automated review tools) both get dramatically simpler for reviewers and contributors.

**Comment by peci1** [comment]
Yeah, something like conda-forge autoprocess would be nice :) They require github usernames only:

https://github.com/conda-forge/staged-recipes/blob/2490b47395996ccf9a560500a6019cd2bb53dc36/recipes/example/meta.yaml#L94-L98

With a github username, I think it could be checked that this user has pushed some commits to the linked repo, basically automating the authorization to become a maintainer (but I'm not sure how to distinguish community PRs from maintainer commits; maybe community never pushes, only commits?).
